### PR TITLE
Support for Limited FIDES__CELERY__* Env Vars

### LINKED ISSUE #2181: Increase code coverage to 85%

### PR DESCRIPTION
**Pull request by pattisdr**
Closes PROD-2181

### Description Of Changes

Add limited support for setting some celery values via Fides Env vars: `FIDES__CELERY__TASK_ALWAYS_EAGER`, `FIDES__CELERY__EVENT_QUEUE_PREFIX`, and `FIDES__CELERY__TASK_DEFAULT_QUEUE`.

### Code Changes

* [ ] Add CelerySettings extending from FidesSettings to add support for 3 specific celery values to be set via specific Fides env vars. Maintains support for setting other variables via toml 

#### Verify selected env vars work 
* [ ] Set env var FIDES__CELERY__TASK_ALWAYS_EAGER=false
* [ ] Bring up fides and worker `nox -s dev -- shell worker`
* [ ] Run an access request - verify that it is executed in the worker, not the fides container:
```shell
[2024-06-12 22:51:55,205: INFO/MainProcess] celery@77e252b74b9c ready.
[2024-06-12 22:53:35,375: INFO/MainProcess] Task fides.api.service.privacy_request.request_runner_service.run_privacy_request[82002a88-42c0-4766-b32c-9b5bf4b8e271] received
2024-06-12 22:53:35.592 | INFO     |  - Dispatching privacy request pri_0d5a9e72-e9b5-4b10-8523-06f55c098395
2024-06-12 22:53:35.694 | INFO     |  - starting traversal
2024-06-12 22:53:35.694 | INFO     |  - Traverse __ROOT__:__ROOT__
2024-06-12 22:53:35.748 | INFO     |  - No results returned for access request pri_0d5a9e72-e9b5-4b10-8523-06f55c098395
2024-06-12 22:53:35.787 | INFO     |  - Filtering Access Request results to return fields associated with data categories
2024-06-12 22:53:35.787 | INFO     |  - Starting access request upload for rule default_access_policy_rule for privacy request pri_0d5a9e72-e9b5-4b10-8523-06f55c098395
2024-06-12 22:53:35.789 | INFO     |  - Writing data to in-memory buffer
2024-06-12 22:53:35.845 | INFO     |  - Privacy request pri_0d5a9e72-e9b5-4b10-8523-06f55c098395 run completed.
[2024-06-12 22:53:35,852: INFO/ForkPoolWorker-2] Task fides.api.service.privacy_request.request_runner_service.run_privacy_request[82002a88-42c0-4766-b32c-9b5bf4b8e271] succeeded in 0.4344340000025113s: None
```

### Verify toml vars still work
* [ ] Override the task default queue in the `fides.toml` file
```
[celery]
task_default_queue = "fides_default_queue_test"
``` 
* [ ] Bring up fides and worker `nox -s dev -- shell worker`
* [ ] In celery, this "fides_default_queue_test" shows up in the list of queues:
```
2024-06-12 22:57:34.203 | INFO     |  - Celery connection setup complete | {'celery_details': {'hostname': 'celery@c063a776a5c7', 'version': '5.2.7 (dawn-chorus)', 'app': 'fides.api.tasks:0xffffa88b4b80', 'transport': 'redis://:**@redis:6379//', 'results': 'redis://:**@redis:6379/', 'concurrency': '2', 'queues': 'fides_default_queue_test|fidesops.messaging|fides.privacy_preferences'}}
```
### Test other celery vars outside of the ones defined for env vars can still be passed in through the toml file
We use redis as our backend, override this with a bad URL:
```
[celery]
result_backend = "bad_backend"
``` 
* [ ] Bring up fides and worker `nox -s dev -- shell worker`
* [ ] Celery worker can't find the bad_backend
```
Traceback (most recent call last):
  File "/opt/fides/lib/python3.10/site-packages/celery/utils/dispatch/signal.py", line 276, in send
    response = receiver(signal=self, sender=sender, **named)
  File "/fides/src/fides/api/worker/__init__.py", line 41, in log_celery_setup
    "results": app.backend.as_uri(),
  File "/opt/fides/lib/python3.10/site-packages/celery/app/base.py", line 1252, in backend
    self._local.backend = new_backend = self._get_backend()
  File "/opt/fides/lib/python3.10/site-packages/celery/app/base.py", line 955, in _get_backend
    backend, url = backends.by_url(
  File "/opt/fides/lib/python3.10/site-packages/celery/app/backends.py", line 69, in by_url
    return by_name(backend, loader), url
  File "/opt/fides/lib/python3.10/site-packages/celery/app/backends.py", line 49, in by_name
    cls = symbol_by_name(backend, aliases)
  File "/opt/fides/lib/python3.10/site-packages/kombu/utils/imports.py", line 59, in symbol_by_name
    module = imp(module_name, package=package, **kwargs)
  File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1004, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'bad_backend'
```

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
  * [ ] if there are any new client scopes created as part of the pull request, remember to update public-facing documentation that references our scope registry
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated
* If there are any database migrations:
  * [ ] Ensure that your downrev is up to date with the latest revision on `main`
  * [ ] Ensure that your `downgrade()` migration is correct and works
    * [ ] If a downgrade migration is not possible for this change, please call this out in the PR description!

### CODE REVIEW

**Comment by pattisdr** [comment]
Thanks so much for offering to review @adamsachs ⭐

File: src/fides/config/__init__.py [hunk 1]
```diff
@@ -69,9 +70,7 @@ class FidesConfig(FidesSettings):
     admin_ui: AdminUISettings
     consent: ConsentSettings
     cli: CLISettings
-    celery: Dict = Field(
-        description="This section can be used to pass config vars to Celery directly.",
-    )
+    celery: CelerySettings
```
Comment (pattisdr) [inline src/fides/config/__init__.py:73]: Other vals set in the toml file can still come in through here I believe because CelerySettings > FidesSettings has extra > allow_all.  These will get passed through to the celery_app.conf

Comment (adamsachs) [reply src/fides/config/__init__.py:73]: thanks for clarifying - i was actually just wondering how those got passed through!

based on your testing notes, it seems like you've tested a "mix" of both .toml file "extra" attributes alongside env vars for our "expected" attributes. is that correct? i'd just want to make sure that's covered (seems like you've covered it). i can also test on my end to confirm :)

**Review by adamsachs — APPROVED** [review]
this looks good to me @pattisdr i don't see any real concerns! i didn't test myself but your testing notes look solid (i just had one clarifying comment in another thread), and the code changes are minor enough that i don't think it's strictly necessary.

if you'd like me to test and try and exercise some edge cases i'd be happy to!

File: src/fides/config/__init__.py (hunk 1 above)
Comment (pattisdr) [reply src/fides/config/__init__.py:73]: I did test a mix, i can see how my instructions didn't spell that out though!

Comment (adamsachs) [reply src/fides/config/__init__.py:73]: heh no your instructions were pretty clear :) i just wanted to be extra sure 👍

File: src/fides/config/celery_settings.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    event_queue_prefix: str = Field(
+        default="fides_worker",
+        description="The prefix to use for event receiver queue names",
+    )
+    task_default_queue: str = Field(
+        default="fides",
+        description="The name of the default queue if a message has no route or no custom queue has been specified",
+    )
+    task_always_eager: bool = Field(
+        default=True,
```
[the remark this answers is not in the document]
Comment (adamsachs) [reply src/fides/config/celery_settings.py:20]: that all makes sense, thanks for clarifying!
